src/components/ItinerariesLayer/ItinerariesLayer.js

```diff
@@ -0,0 +1,43 @@
+import { ArcLayer } from '@peripleo/peripleo/deck.gl';
+
+const toArc = (waypoints, graph) => {
+  const trajectory = [];
+
+  for (let i=1; i<waypoints.length; i++) {
+    trajectory.push({
+      from: graph.getNode(waypoints[i-1].place).geometry,
+      to: graph.getNode(waypoints[i].place).geometry
+    });
+  }
+
+  return trajectory;
+}
+
+/**
+ * The task of the ItineraryLayer is to take a search result,
+ * and turn the itinerary trajectories into DeckGL layers.
+ */
+export const ItinerariesLayer = (resultItems, graph) => {
```

Comment:
How much effort to also include a GeoJSON layer of each linked place entry that is in any of the filtered itineraries? Do you think that would be a performance bottleneck with this structure? (Can split out into a separate issue/branch/PR, btw)

Reply:
Perfect, thanks! I will open a new issue for it. I'd actually like to try implementing it myself so I can get a better understanding of how to work with your new code 😄